airbyte-integrations/connectors/source-zuora/source_zuora/source.py

```diff
@@ -0,0 +1,166 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+from abc import ABC
+from typing import Any, Dict, Iterable, List, Mapping, MutableMapping, Tuple
+
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.models import AirbyteStream
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+
+from .zuora_auth import ZuoraAuthenticator
+from .zuora_client import ZoqlExportClient
+from .zuora_errors import ZOQLQueryCannotProcessObject, ZOQLQueryFieldCannotResolve
+
+
+# Main class for Zuora Stream with overriden CDK methods
+class ZuoraStream(Stream, ABC):
+    # Define general primary key
+    primary_key = "id"
+
+    def __init__(self, api: ZoqlExportClient):
+        self.api = api
+
+    def get_cursor_from_schema(self, schema: Dict) -> str:
+        """
+        Get the cursor_field from the stream's schema,
+        If the stream doesn't support 'updateddate', then we use 'createddate'.
+        """
+        return self.alt_cursor_field if self.cursor_field not in schema.keys() else self.cursor_field
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        """
+        Get the stream's schema from Zuora Object's Data types,
+        """
+        schema = {}
+        schema["properties"] = self.api._zuora_object_to_json_schema(self.name)
```

Comment:
```suggestion
        return {"properties": self.api._zuora_object_to_json_schema(self.name)}
```

Reply:
Will check if this works as expected, thanks

Reply:
> "$schema": "http://json-schema.org/draft-07/schema#",
> "type": "object",

I didn't notice any issues without it, the CATALOG is built properly and normalization is ok as well.